### ISSUE TITLE
Unable to login due to not redirecting from browser

### ISSUE THREAD

**Issue by ynakao** [QuantumBadger/RedReader#1100]
I upgraded RedReader v1.21.1 on f-droid yesterday and the app prompted re-authentication, so I tried re-login but re-auth process stalled without redirecting to the app after login from the internal web browser.

I found related discussions on /r/redreader subreddit [[1]](https://reddit.com/r/RedReader/comments/14lagej/unable_to_add_account_browser_does_not_redirect/), [[2]](https://reddit.com/r/RedReader/comments/14ns6zl/version_121_released/jq96kvv/) and [the issue tracker](https://github.com/QuantumBadger/RedReader/issues/985). @QuantumBadger said the issue was fixed in those threads, but it still affects me.

Deleting cache and reinstalling RedReader and revoking the app permission on my reddit preference don't work. Any suggestions? I'm using Android 7.1.2.

**Comment by Blizihguh** [comment]
Having the same issue (or at least I think so). I log in on using the app's internal browser, it logs in, then says "Error fetching oauth client" when redirecting. I'm on the most recent android version for my device (much higher than 4.x) and tried the workaround from the closed issue, so none of the links in the OP apply to my situation. That being said, I did build the app from source, so it's possible that I did something wrong. (I did set my API key at build time!)

**Comment by QuantumBadger** [comment]
@ynakao Thanks for the report. There's a limited amount I can do because Reddit handles the login flow on their site, however I can report to Reddit that the issue is still occurring. Would it be possible to send a video of the issue please, starting at the point after you've typed in your password?

@Blizihguh That sounds like a different issue -- you may be using an invalid API key, or it might be in the wrong place. You could try setting it in `Settings > Network > Reddit client ID override` instead.

**Comment by Blizihguh** [comment]
Yep, can confirm it was an issue on my end. Specifically, I was able to fix it by setting the correct redirect URI for my token ("redreader://rr_oauth_redir"). Thanks for pointing me in the right direction! I think that info should probably be added to CONTRIBUTING.md, since it took some digging to find the root cause and the correct URI to supply.

**Comment by ynakao** [comment]
@QuantumBadger Here you are. About 1min 2MB clip.

I expected redirection occurred after I accepted the app permission, but nothing happened. Instead, `Sign Up or Log In` message in a blue banner appeared at around 15s in video. Tapped the button to continue, but it only led to the Reddit web view in the internal browser, and my account was not listed on the RedReader accounts list.

https://github.com/QuantumBadger/RedReader/assets/6405335/3078d64d-1d28-4238-a402-2f3806c90e1f

**Comment by QuantumBadger** [comment]
Thanks both!

@ynakao I've passed this video on to Reddit, they've been looking into it for a while but no results yet.

In the meantime, @nascentt suggested trying to upgrade your system WebView:

https://old.reddit.com/r/RedReader/comments/14z8ear/unable_to_connect_account/

**Comment by ynakao** [comment]
As a workaround, I downgraded RedReader to v1.20 and re-authenticated successfully in the external browser(fennec) following [the reddit comment](https://old.reddit.com/r/RedReader/comments/15cxgjz/login_impossible_now/ju2p5b5/).

**Comment by Helyos96** [comment]
Installing a recent android system webview did the trick for me. However I had to go to developer options to switch to it, otherwise it was stuck to the one that shipped with the ROM.

**Comment by SpitFire-666** [comment]
Likewise, updating Android Webview fixed it for me. Perhaps RedReader could allow users to choose their own browser in future for the authing? I think Firefox offers a webview alternative.

**Comment by Helyos96** [comment]
> So, does this issue stem from the old WebView implementation? Can this be resolved by either RedReader side or Reddit?

I think this is reddit's login site which is no longer compatible with older browsers. Not much RR can do imho.

**Comment by binamah** [comment]
@QuantumBadger Would it be possible to allow authenticate with external browser. Comment by https://github.com/QuantumBadger/RedReader/issues/1100#issuecomment-3395423491  works fine
